File: app/src/main/java/org/getlantern/lantern/model/AccountInitializationStatus.java
package org.getlantern.lantern.model;

public class AccountInitializationStatus {

    private AccountInitializationStatus.Status status;

    public enum Status {
        PROCESSING, SUCCESS, FAILURE;
    }

    public AccountInitializationStatus(final AccountInitializationStatus.Status status) {
        this.status = status;
    }

    public AccountInitializationStatus.Status getStatus() {
        return status;
    }

    public boolean isProcessing() {
        return status != null && status.equals(Status.PROCESSING);
    }

    public boolean isSuccess() {
        return status != null && status.equals(Status.SUCCESS);
    }

    public boolean isFailure() {
        return status != null && status.equals(Status.FAILURE);
    }

}

File: android/app/src/main/kotlin/org/getlantern/lantern/model/AccountInitializationStatus.kt
package org.getlantern.lantern.model

data class AccountInitializationStatus(val status: Status) {

    enum class Status {
        PROCESSING, SUCCESS, FAILURE
    }

    fun isProcessing(): Boolean {
        return status == Status.PROCESSING
    }

    fun isSuccess(): Boolean {
        return status == Status.SUCCESS
    }

    fun isFailure(): Boolean {
        return status == Status.FAILURE
    }
}
